### PR TITLE
feat: add bayesian elo model to trinomial SPRT and improve LLR formula

### PR DESCRIPTION
**Pull request by gahtan-syarif**
does two things:
1. introduce bayesian model to trinomial sprt, which is the original model that fishtest and openbench uses before the introduction of pentanomial sprt
2. use an improved LLR formula by vdbergh that improves LLR accuracy when elo differences are huge: https://talkchess.com/forum/viewtopic.php?t=57465&start=27

### CODE REVIEW

**Comment by gahtan-syarif** [comment]
i have tested the output and it produced the expected results, i believe this can be merged now

File: src/types/tournament_options.hpp [hunk 1]
```diff
@@ -39,9 +39,9 @@ struct Sprt {
     double beta          = 0.0;
     double elo0          = 0.0;
     double elo1          = 0.0;
-    bool logistic_bounds = false;
+    std::string model    = "normalized";
```
Comment (Disservin) [inline src/types/tournament_options.hpp:42]: can you document the available models as a comment here please?

**Comment by Disservin** [comment]
> i have tested the output and it produced the expected results, i believe this can be merged now

can you add the test cases or specify them here=

File: src/types/tournament_options.hpp (hunk 1 above)
Comment (gahtan-syarif) [reply src/types/tournament_options.hpp:44]: added 👍

**Comment by gahtan-syarif** [comment]
> > i have tested the output and it produced the expected results, i believe this can be merged now
> 
> can you add the test cases or specify them here=

here are some test cases we can use:
sprt test case results (rounded to 2 decimals):
-normalized model:
(case 1)
bounds=[0,2]
Total: 141152 W: 36433 L: 36027 D: 68692
Ptnml(0-2): 365, 16618, 36229, 16974, 390
pentanomial llr=2.25
trinomial llr=0.92
(case 2)
bounds=[-1.75,0.25]
Total: 41952 W: 10871 L: 10650 D: 20431
Ptnml(0-2): 127, 4883, 10712, 5150, 104
pentanomial llr=3.01
trinomial llr=2.30

-logistic model:
(case 1)
bounds=[0.5,2.5]
Total: 83296 W: 21404 L: 21184 D: 40708
Ptnml(0-2): 223, 9863, 21279, 10037, 246
pentanomial llr= -3.07
trinomial llr= -1.57
(case 2)
bounds=[0,2]
Total: 221056 W: 57433 L: 57030 D: 106593
Ptnml(0-2): 871, 26175, 55983, 26678, 821
pentanomial llr= -4.98
trinomial llr= -2.59

-bayesian model:
(case 1)
bounds=[0,2]
Total: 265920 W: 68965 L: 68526 D: 128429
trinomial llr= -1.26
(case 2)
bounds=[0.5,2.5]
Total: 84224 W: 21629 L: 21484 D: 41111
trinomial llr= -1.13

**Comment by gahtan-syarif** [comment]
added comment for the source of the formula as well

**Comment by gahtan-syarif** [comment]
ci bugging out

**Comment by gahtan-syarif** [comment]
added test cases for sprt

**Pull request merged**